packaging.md

```diff
@@ -0,0 +1,156 @@
+
+# copy the source to the docker image
+COPY . /workspace
+
+RUN swift build -c release -Xswiftc -g
+
+#------- package -------
+# Running on distroless C++ since it includes
+# all(*) the runtime dependencies Swift programs need
+FROM gcr.io/distroless/cc-debian10
+# copy executables
+COPY --from=builder /workspace/.build/release/<executable-name> /
+# copy Swift's dynamic libraries dependencies
+COPY --from=builder /usr/lib/swift/linux/lib*so* /
+
+# set the entry point (application name)
+CMD ["<executable-name>"]
+```
+
+Note the above uses `gcr.io/distroless/cc-debian10` as the runtime image which should work for Swift programs that do not use `FoundationNetworking` or `FoundationXML`. In order to provide more complete support we (the community) could put in a PR into distroless to introduce a base image for Swift that includes `libcurl` and `libxml` which are required for `FoundationNetworking` and `FoundationXML` respectively.
+
+## Archive (Tarball, ZIP file, etc.)
+
+Since cross-compiling Swift for Linux is not (yet) supported on Mac or Windows, we need to use virtualization technologies like Docker to compile applications we are targeting to run on Linux.
+
+That said, this does not mean we must also package the applications as Docker images in order to deploy them. While using Docker images for deployment is convenient and popular, an application can also be packaged using a simple and lightweight archive format like tarball or ZIP file, then uploaded to the server where it can be extracted and run.
+
+Here is an example of using Docker and `tar` to build and package the application for deployment on Ubuntu servers:
+
+First, use the `docker run` command from the application's source location to build it:
+
+```bash
+$ docker run --rm \
+  -v `pwd`:/workspace \
+  -w /workspace \
+  swift:5.2-bionic \
+  /bin/bash -cl "swift build -c release -Xswiftc -g"
+```
+
+Note we are bind mounting the source directory so that the build writes the build artifacts to the local drive from which we will package them later.
+
+Next we can create a staging area with the application's executables and Swift's dynamic libraries dependencies:
+
+```bash
+$ docker run --rm \
+  -v `pwd`:/workspace \
+  -w /workspace \
+  swift:5.2-bionic  \
+  /bin/bash -cl ' \
+     rm -rf .build/install && mkdir -p .build/install; \
+     cp -P .build/release/<executable-name> .build/install/; \
+     cp -P /usr/lib/swift/linux/lib*so* .build/install/'
+```
+
+Note this command could be combined with the build command above--we separated them to make the example more readable.
+
+Finally, create a tarball from the staging directory:
+
+```bash
+$ tar cvzf <my-app>-<my-app-version>.tar.gz -C .build/install .
+```
+
+We can test the integrity of the tarball by extracting it to a directory and running the application in a Docker runtime container:
+
+```bash
+$ cd <extracted directory>
+$ docker run -v `pwd`:/app -w /app swift:5.2-bionic-slim ./<executable-name>
```

Comment:
- I think we should quote this just in case the dir has a space
- I think we shouldn't recommend the outdated ``` ` ```

So what about

    docker run -v "$(pwd)":/app -w /app swift:5.2-bionic-slim ./<executable-name>

Personally, I actually use

    docker run -v "$PWD:/app" -w /app swift:5.2-bionic-slim ./<executable-name>

Reply:
sure. my age shows :)